cupy/fft/_fft.py

```diff
@@ -253,6 +255,11 @@ def _fft(a, s, axes, norm, direction, value_type='C2C', overwrite_x=False,
         out_size = _get_fftn_out_size(a.shape, s, axes[-1], value_type)
         a = _exec_fft(a, direction, value_type, norm, axes[-1], overwrite_x,
                       out_size)
+        # Numpy compatibility
+        # 1D FFT: Regain the original type (before _convert_dtype)
+        # Others: Not necessary as numpy applies C2C iFFTs before R2C iRFFT
+        if len(axes) == 1:
+            a = a.astype(real_dtype)
```

Comment:
For performance.
```suggestion
            a = a.astype(real_dtype, copy=None)
```

Reply:
Fixed in 0afe0cf7d266559e47260afb94c51a4279acbbfe .